draft-irtf-cfrg-hpke.md

```diff
@@ -428,37 +470,40 @@ def KeySchedule(mode, pkR, zz, enc, info, psk, pskID, pkSm):
   VerifyMode(mode, psk, pskID, pkSm)
 
   pkRm = Marshal(pkR)
-  identifier = "RFCXXXX"
   ciphersuite = concat(encode_big_endian(kem_id, 2),
                        encode_big_endian(kdf_id, 2),
                        encode_big_endian(aead_id, 2))
-  pskID_hash = Hash(pskID)
-  info_hash = Hash(info)
+  pskID_hash = Extract(0, concat("pskID", pskID))
+  info_hash = Extract(0, concat("info", info))
+  identifier = "RFCXXXX HPKE"
   context = concat(identifier, ciphersuite, mode, enc, pkRm,
                    pkSm, pskID_hash, info_hash)
 
-  secret = Extract(psk, zz)
+  if length(psk) > Nb then:
+    psk = Extract(0, concat("psk", psk))
+
+  secret = Extract(psk, concat("zz", zz))
   key = Expand(secret, concat("key", context), Nk)
   nonce = Expand(secret, concat("nonce", context), Nn)
   exporter_secret = Expand(secret, concat("exp", context), Nh)
 
   return Context(key, nonce, exporter_secret)
 ~~~~~
-
-\[\[RFC editor: please change "RFCXXXX" to the correct number before
-publication.]]
+\[\[RFC editor: please change "RFCXXXX" to the correct number before publication.]]
 
 Note that the context construction in the KeySchedule procedure is
 equivalent to serializing a structure of the following form in the
 TLS presentation syntax:
 
 ~~~~~
 struct {
+    uint8 identifier[12];
+
```

Comment:
Is this correct for the 12-letter UTF-8 string literal “RFCXXXX HPKE”?

Reply:
Let's treat it as an ASCII string, in which case, yes!